File: src/main/java/com/webatrio/test/service/EventsService.java
package com.webatrio.test.service;

import com.webatrio.test.models.Events;
import com.webatrio.test.models.User;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Set;

public interface EventsService {
    Events addEvent(Events events);
    Events getEventById(Long id);
    Events cancelEvent(Long eventId);
    Events addUserToEvent(Long userId, Long eventId);
    Events deleteUserFromEvent(Long userId, Long eventId);
    Page<Events> findEvents(Integer page, Integer size);
    Page<Events> findEventsBylieu(String lieu, Integer page, Integer size);
    List<User> findUsersByEvent(Long id);
    Page<Events> getEventsForUser(Long userId, Integer page, Integer size);
}

File: angular-11-client/src/app/_services/events.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
const apiUrl = 'http://localhost:8085/api/events';
@Injectable({
  providedIn: 'root'
})
export class EventsService {

  constructor(private http: HttpClient) { }



  getEvents(page: number, size: number): Observable<any> {
    const paginationParams = {
      page: page,
      size: size
    };
    return this.http.post<any>(apiUrl, paginationParams)
      .pipe(
      );
  }

  getMyEvents(userid: number,paginationParams: any): Observable<any> {
    return this.http.post<any>(apiUrl+`/findEventsForUser/${userid}`, paginationParams)
      .pipe(
      );
  }
  addEvent(event: any): Observable<any> {
    return this.http.post<any>(apiUrl+'/add', event);
  }
  cancelEvent(id: number): Observable<any> {
    return this.http.put<any>(apiUrl+`/cancel/${id}`, {});
  }


  getEventsByLieu(lieu: string, paginationParams: any): Observable<any> {
    return this.http.post(apiUrl+`/byLieu/${lieu}`, paginationParams);
  }

  addUserToEvent(idUser: number, idEvent: number): Observable<any> {
    return this.http.put<any>(apiUrl+`/addUserToEvent/${idUser}/${idEvent}`, {});
  }
  deleteUserFromEvent(idUser: number, idEvent: number): Observable<any> {
    return this.http.put<any>(apiUrl+`/deleteUserFromEvent/${idUser}/${idEvent}`, {});
  }
  getUserByEvents(eventid: number,paginationParams: any): Observable<any> {
    return this.http.post<any>(apiUrl+`/findUsersByEvent/${eventid}`, paginationParams)
      .pipe(
      );
  }
}

